### ISSUE TITLE
Feat: BUG: transaction preparation fails for non-account-deploying transactions on custom networks #2090

### ISSUE THREAD

**Issue by Cevedale** [argentlabs/argent-x#2096]
Update the extension or the library that manages the Starknet provider to a version that supports the RPC operation. Version (v4) is deprecated and not suitable for the RPC call. Consult the official documentation or release notes for the extension or library to find the latest version that supports RPC.

Check if the Starkli command-line tool is up to date. You can update it using the following command:

bash
Copy code
npm install -g starkware-cli
This assumes you have Node.js and npm installed. If not, you can install them from https://nodejs.org/en .

Verify that the network configuration (Katana) is correctly set up. Ensure that the RPC endpoint is accessible and the network details (chain ID, RPC URL) match the actual configuration of the Katana network. Sometimes, these details can change, and using outdated configurations can lead to issues.

Don't forget to update also the Starknet extension.

**Comment by 9oelM** [comment]
Yes, can totally replicate. My team also has a full local dev env set up with Katana. Shows something like this. 
![image](https://github.com/argentlabs/argent-x/assets/22465806/da0ae373-75ed-47ff-8599-64ffd2561fca)

The error message `40: Contract error: undefined` is not quite helpful actually.

**Comment by 9oelM** [comment]
Yeap downgrading to [v5.10.1](https://github.com/argentlabs/argent-x/releases/tag/v5.10.1) works. Download the .zip from GH release page and use 'load unpacked' on Chrome to load the extension. Didn't find out the root cause yet though. But well we know we have a temporary solution now.
